### PR TITLE
[integ-tests][OSU] Use rolling window for benchmark failure detection

### PR DESCRIPTION
**Pull request by hanwen-cluster**
### Description of changes
Replace the fixed threshold count (3 for collective, 4 for pt2pt) with a rolling window approach: a benchmark is flagged as failed only if 3 out of any 4 consecutive packet sizes exceed the tolerance.

The old method counted total failures across all packet sizes, treating scattered failures the same as clustered ones. Since benchmark characteristics vary by packet-size range, scattered failures are more likely noise while clustered failures indicate a real regression in a specific range.

Simulated against the 5 most recent DynamoDB runs with the new p75 thresholds:

Old detection: 10/30 test failures (33.3%)
Rolling window 4/3: 8/30 test failures (26.7%)
The 2 eliminated false positives had scattered failures across unrelated packet sizes. The 8 remaining failures all have dense clusters (e.g. 3+ consecutive sizes at ps 4-32 in osu_allreduce), which are stronger signals of actual issues.

### Tests
tests on develop.yaml have passed

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/performance_tests/test_osu.py [hunk 1]
```diff
@@ -178,10 +176,10 @@ def _test_osu_benchmarks_pt2pt(
             network_interfaces_count,
             test_datadir,
         )
-        failures = _check_osu_benchmarks_results(
+        failure_mask = _check_osu_benchmarks_results(
```
Comment (gmarciani) [inline tests/integration-tests/tests/performance_tests/test_osu.py:179]: [nitpick] failure mask is misleading. Those are actual failures. We then process them to identify clustered failures

File: tests/integration-tests/tests/performance_tests/test_osu.py [hunk 2]
```diff
@@ -366,4 +362,25 @@ def _check_osu_benchmarks_results(
...
+    Instead of counting total failures across all packet sizes, look for clusters
+    of failures in consecutive packet sizes. This catches real regressions in a
+    specific packet-size range while ignoring scattered noise.
+
+    A benchmark is considered failed if any window of `window_size` consecutive
+    packet sizes contains at least `min_failures` failures.
+    """
+    if not failure_mask:
+        return False
+    if len(failure_mask) < window_size:
```
Comment (gmarciani) [inline tests/integration-tests/tests/performance_tests/test_osu.py:381]: if the number of failures is less than the window, then it should return False.

File: tests/integration-tests/tests/performance_tests/test_osu.py [hunk 3]
```diff
@@ -366,4 +362,25 @@ def _check_osu_benchmarks_results(
...
+        return sum(failure_mask) >= min(min_failures, len(failure_mask))
+    for i in range(len(failure_mask) - window_size + 1):
```
Comment (gmarciani) [inline tests/integration-tests/tests/performance_tests/test_osu.py:383]: this logic assumes that the content of failure_mask is an array of elements 0,1 where 0 is success and 1 is failure, but I think this is not what _check_osu_benchmarks_results returns. Can we double check?

File: tests/integration-tests/tests/performance_tests/test_osu.py (hunk 1 above)
Comment (hanwen-cluster) [reply tests/integration-tests/tests/performance_tests/test_osu.py:179]: The structure of failure_mask is [0,1,1,0,0]. So it is indeed masks not real failures

File: tests/integration-tests/tests/performance_tests/test_osu.py (hunk 2 above)
Comment (hanwen-cluster) [reply tests/integration-tests/tests/performance_tests/test_osu.py:381]: Here is a corner case, when tests run on fewer packet sizes than the window size. For example, barrier test only has one entry, one failure should be reported

File: tests/integration-tests/tests/performance_tests/test_osu.py (hunk 3 above)
Comment (hanwen-cluster) [reply tests/integration-tests/tests/performance_tests/test_osu.py:383]: Logic is here https://github.com/aws/aws-parallelcluster/pull/7281/changes#diff-60b99ffc235e925350dc9f5ce1b9580076a5e5910a4638b037c68810fc3780afR353

Comment (gmarciani) [reply tests/integration-tests/tests/performance_tests/test_osu.py:383]: Got it, so indeed it is a failure mask :-D

**Review by gmarciani — APPROVED** [review]
